### PR TITLE
Bump argo submodule to include docs for method results

### PR DESCRIPTION
**Pull request by RyanGlScott**
This follows up on a promise I made in https://github.com/GaloisInc/cryptol/issues/1196#issuecomment-841179362 to bump the `argo` submodule to include `argo` commit https://github.com/GaloisInc/argo/commit/b9b3edd992e57ca31224530f763f313fcf24bab7, which adds the ability to document the results of methods (in addition to their parameters). In addition to dealing with the breaking API changes from that commit, this adds missing documentation for each method's results.

Along the way, I realized that `cryptol-remote-api`'s documentation is hand-written rather than autogenerated (see #1206). In pursuit of eventually fixing that issue, the CI now makes sure that any changes to the autogenerated documentation are checked in.

### CODE REVIEW

File: cryptol-remote-api/check_docs.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+cd $DIR/docs
+
+export CRYPTOL_SERVER=$(which cryptol-remote-api)
```
Comment (pnwamk) [inline cryptol-remote-api/check_docs.sh:7]: The other existing RPC script uses a [slightly different method](https://github.com/GaloisInc/cryptol/blob/1d8a2fdb527bc85b0aa7acbf8a0d819a26262184/cryptol-remote-api/run_rpc_tests.sh#L24) to find the server:
```
export CRYPTOL_SERVER=$(cabal v2-exec which cryptol-remote-api)
```
Should these be the same and/or more similar? (I'm just imagining one failing and the other not and that being kind of funny/frustrating.) I think personally I have usually relied on `cabal run` or `cabal exec`, but I certainly don't know what method for finding the executable is the "best".

Comment (RyanGlScott) [reply cryptol-remote-api/check_docs.sh]: I'll be honest—I cargo-culted much of this script from [`saw-script`'s counterpart](https://github.com/GaloisInc/saw-script/blob/5535b03531d0790cddb01d732be316c96838a8db/saw-remote-api/scripts/check_docs.sh#L9) without thinking too deeply about it. But yes, we should be consistent with the conventions already established in `cryptol`. Fixed.

Comment (RyanGlScott) [reply cryptol-remote-api/check_docs.sh]: And as luck would have it, that's exactly the change I needed to make anyway in order to fix CI :)

**Review by pnwamk — APPROVED** [review]
LGTM!

While perusing the docs I discovered some issues but I think they're probably just better addressed separately instead of holding up this PR.

In particular, some of `Argo`'s docs need updating: https://github.com/GaloisInc/argo/pull/166

And perhaps it's worth considering generating something more polished from the `.rst`: https://github.com/GaloisInc/cryptol/issues/1208

**Pull request merged**